CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+# Contributing to Trust-DNS
+
+Before contributing, please consider the terms of the licenses (Apache License 2.0 or MIT). We chose these licenses for two reasons:
+
+- To be more compatible with the general Rust ecosystem
+- So that this software can be liberally used with few restrictions
+
+After ensuring the license options are compatible with the aims of the contribution, then please submit your PR for review and we will review as soon as possible. My only ask is that you do not do this for free, unless it's something that is a passion or learning project for you. Please, find a way to be paid for your work. You are worth it.
+
+## Understanding the design
+
+Please read the [Architecture](ARCHITECTURE.md) to understand the general design and layout of the Trust-DNS projects.
+
+## Submitting PRs
+
+Before submitting a PR it would be good to discuss the change in an issue so as to avoid wasted work, also feel free to reach out on the Discord channel listed on the front page of the GitHub project. Please, consider keep PRs focused on one issue at a time. While issues are not required for a PR to be accepted they are encouraged, especially for anything that would change behavior, change an API, or be a medium to large change.
+
+If changes will span multiple crates, consider breaking up the PR into multiple commits such that a reviewer can follow the change. This is no necessary, but can make it easier for a reviewer to follow the changes.
```

Comment:
I'm not sure spanning multiple crates is the important metric here. I recently saw this expressed as "refactoring commits should be separate from functional change commits", which seems like a nice pithy way of putting it.